spanconfig/spanconfigsqlwatcher: TestSQLWatcherOnEventError failed

spanconfig/spanconfigsqlwatcher.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3881135&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3881135&tab=artifacts#/) on master @ [5509f95002044013d9164bf8340104051cee93f1](https://github.com/cockroachdb/cockroach/commits/5509f95002044013d9164bf8340104051cee93f1):


```
I211210 02:36:31.580896 7717 1@server/server.go:1594  [n1] 223  connecting to gossip network to verify cluster ID "88599f51-8cd6-4738-a760-cd5e66beefd7"
W211210 02:36:31.580846 7618 1@gossip/gossip.go:1472  [n1] 224  no addresses found; use --join to specify a connected node
I211210 02:36:31.581066 7618 gossip/gossip.go:401  [n1] 225  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:37643" > attrs:<> locality:<tiers:<key:"region" value:"test" > tiers:<key:"dc" value:"dc1" > > ServerVersion:<major_val:21 minor_val:2 patch:0 internal:26 > build_tag:"v22.1.0-alpha.00000000-1960-g5509f95002" started_at:1639103791581058921 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:39947" >
I211210 02:36:31.588788 7618 server/node.go:450  [n1] 226  initialized store s1
I211210 02:36:31.588979 7618 kv/kvserver/stores.go:250  [n1] 227  read 0 node addresses from persistent storage
I211210 02:36:31.589155 7618 server/node.go:527  [n1] 228  started with engine type 2
I211210 02:36:31.589224 7618 server/node.go:529  [n1] 229  started with attributes []
I211210 02:36:31.590343 7717 1@server/server.go:1597  [n1] 230  node connected via gossip
I211210 02:36:31.591390 7618 1@server/server.go:1725  [n1] 231  starting https server at 127.0.0.1:37083 (use: 127.0.0.1:37083)
I211210 02:36:31.591495 7618 1@server/server.go:1730  [n1] 232  starting postgres server at 127.0.0.1:39947 (use: 127.0.0.1:39947)
I211210 02:36:31.591545 7618 1@server/server.go:1732  [n1] 233  starting grpc server at 127.0.0.1:37643
I211210 02:36:31.591595 7618 1@server/server.go:1733  [n1] 234  advertising CockroachDB node at 127.0.0.1:37643
I211210 02:36:31.611330 7618 1@util/log/event_log.go:32  [n1] 235 ={"Timestamp":1639103791611325220,"EventType":"node_join","NodeID":1,"StartedAt":1639103791581058921,"LastUp":1639103791581058921}
I211210 02:36:31.611420 7618 sql/sqlliveness/slinstance/slinstance.go:296  [n1] 236  starting SQL liveness instance
I211210 02:36:31.621111 7709 sql/sqlliveness/slstorage/slstorage.go:403  [n1] 237  inserted sqlliveness session eb2e1ab95afd4cc8a6e1bf76d1ef3722
I211210 02:36:31.621180 7709 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 238  created new SQL liveness session eb2e1ab95afd4cc8a6e1bf76d1ef3722
I211210 02:36:31.623873 7882 kv/kvserver/replica_rangefeed.go:648  [n1,rangefeed=table-stats-cache,s1,r16/1:/Table/2{0-1}] 239  RangeFeed closed timestamp is empty
I211210 02:36:31.646698 7936 sql/temporary_schema.go:513  [n1] 240  running temporary object cleanup background job
I211210 02:36:31.647469 7936 sql/temporary_schema.go:557  [n1] 241  found 0 temporary schemas
I211210 02:36:31.648788 7936 sql/temporary_schema.go:560  [n1] 243  early exiting temporary schema cleaner as no temporary schemas were found
I211210 02:36:31.648947 7936 sql/temporary_schema.go:561  [n1] 244  completed temporary object cleanup job
I211210 02:36:31.649107 7936 sql/temporary_schema.go:644  [n1] 245  temporary object cleaner next scheduled to run at 2021-12-10 03:06:31.646646626 +0000 UTC
I211210 02:36:31.648639 8034 sql/sqlstats/persistedsqlstats/provider.go:137  [n1] 242  starting sql-stats-worker with initial delay: 53m15.90177561s
I211210 02:36:31.734875 8087 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 246 ={"Timestamp":1639103791714480703,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I211210 02:36:31.740237 8210 migration/migrationmanager/manager.go:110  [n1,intExec=set-setting,migration-mgr] 247  no need to migrate, cluster already at newest version
I211210 02:36:31.742302 8210 util/log/event_log.go:32  [n1,intExec=set-setting] 248 ={"Timestamp":1639103791738686958,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.2-26'"],"SettingName":"version","Value":"21.2-26"}
I211210 02:36:31.780436 8486 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 249 ={"Timestamp":1639103791772943913,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"4dcd38c1-4242-48ef-8657-1e625e4a91c7"}
I211210 02:36:31.807586 8586 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 250 ={"Timestamp":1639103791802379837,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I211210 02:36:31.847459 7728 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 251  [JOB 3] WAL created 000004
I211210 02:36:31.857159 8592 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 252 ={"Timestamp":1639103791807974086,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":52,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I211210 02:36:31.948925 7618 server/server_sql.go:1130  [n1] 253  done ensuring all necessary startup migrations have run
I211210 02:36:31.957094 8840 jobs/job_scheduler.go:357  [n1] 254  waiting 2m0s before scheduled jobs daemon start
I211210 02:36:31.964228 8842 server/auto_upgrade.go:55  [n1] 255  no need to upgrade, cluster already at the newest version
I211210 02:36:31.980653 7618 1@server/server.go:2417  [n1] 256  serving sql connections
I211210 02:36:31.981393 7618 testutils/testcluster/testcluster.go:1158  [-] 257  WaitForFullReplication
I211210 02:36:31.981436 7618 testutils/testcluster/testcluster.go:1162  [-] 258  WaitForFullReplication took: 349ns
I211210 02:36:31.981492 7618 2@rpc/context.go:1103  [n1] 259  dialing n1: 127.0.0.1:37643 (default)
I211210 02:36:32.075984 7981 util/log/event_log.go:32  [n1,client=127.0.0.1:34406,hostssl,user=root] 260 ={"Timestamp":1639103792060982850,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.rangefeed.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.rangefeed.enabled","Value":"true"}
I211210 02:36:32.109265 7981 util/log/event_log.go:32  [n1,client=127.0.0.1:34406,hostssl,user=root] 261 ={"Timestamp":1639103792077253731,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.closed_timestamp.target_duration\" = '100ms'","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.closed_timestamp.target_duration","Value":"00:00:00.1"}
I211210 02:36:32.118823 7981 5@util/log/event_log.go:32  [n1,client=127.0.0.1:34406,hostssl,user=root] 262 ={"Timestamp":1639103792110199056,"EventType":"create_table","Statement":"CREATE TABLE defaultdb.public.t ()","Tag":"CREATE TABLE","User":"root","DescriptorID":54,"TableName":"defaultdb.public.t"}
I211210 02:36:32.122776 5725 kv/kvserver/replica_command.go:400  [n1,split,s1,r43/1:/{Table/47-Max}] 263  initiating a split of this range at key /Table/54 [r44] (span config)
I211210 02:36:32.132764 7618 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:264  [-] 264  established range feed over system.descriptors table starting at time 1639103792.109942953,0
I211210 02:36:32.132849 7618 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:320  [-] 265  established range feed over system.zones table starting at time 1639103792.109942953,0
    sqlwatcher_test.go:367: handler should never run
I211210 02:36:32.231012 7618 testutils/testcluster/testcluster.go:106  [-] 266  TestCluster quiescing nodes
W211210 02:36:32.231166 8070 jobs/registry.go:851  [n1] 267  canceling all adopted jobs due to stopper quiescing
W211210 02:36:32.231462 9234 kv/kvserver/intentresolver/intent_resolver.go:578  [-] 268  failed to cleanup transaction intents: failed to resolve intents: kv/kvserver/intentresolver/intent_resolver.go:901: stopping
I211210 02:36:32.231672 8883 2@rpc/context.go:742  [n1] 269  closing [{n1: 127.0.0.1:37643 (default)} {n0: 127.0.0.1:37643 (default)}]
W211210 02:36:32.231898 7709 sql/sqlliveness/slinstance/slinstance.go:220  [n1] 270  exiting heartbeat loop
--- FAIL: TestSQLWatcherOnEventError (0.83s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/spanconfig/spanconfigsqlwatcher/sqlwatcher_test.go
+++ b/pkg/spanconfig/spanconfigsqlwatcher/sqlwatcher_test.go
@@ -345,7 +345,7 @@ func TestSQLWatcherOnEventError(t *testing.T) {
 	tdb.Exec(t, `SET CLUSTER SETTING kv.rangefeed.enabled = true`)
 	tdb.Exec(t, `SET CLUSTER SETTING kv.closed_timestamp.target_duration = '100ms'`)
 
-	noopCheckpointDuration := 100 * time.Millisecond
+	noopCheckpointDuration := time.Hour // effectively disable no-op checkpoints
 	sqlWatcher := spanconfigsqlwatcher.New(
 		keys.SystemSQLCodec,
 		ts.ClusterSettings(),

